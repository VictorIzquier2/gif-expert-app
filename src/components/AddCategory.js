import React, {Fragment, useState} from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {

  const [inputValue, setInputValue] = useState('');

  const handleInputChange = (e) => {
    setInputValue(e.target.value)

    console.log('handleInputChange llamado');
  }

  const handleSubmit = (e) => {
    e.preventDefault();

    console.log('handleSubmit', inputValue);

    if(inputValue.trim().length > 0){
      setCategories(cats => [inputValue, ...cats])
      setInputValue('');
    }
  }

  return (
    <Fragment>
      <form
        onSubmit={handleSubmit}
      >
        <p>{inputValue}</p>
        <input
          type='text'
          value={inputValue}
          onChange={handleInputChange}
        />
      </form>
    </Fragment>
   );
}
AddCategory.protoTypes = {
  setCategories: PropTypes.func.isRequired
}
 
export default AddCategory;