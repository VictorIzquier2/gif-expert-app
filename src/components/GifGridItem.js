import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

export const GifGridItem = ({url, title}) => {
  return ( 
    <Fragment>
      <div className='card'>
        <img
          src={url}
          alt={title}
          height={240}
        />
        <p>{title}</p>
      </div>
    </Fragment>
   );
}

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
}
export default GifGridItem;
