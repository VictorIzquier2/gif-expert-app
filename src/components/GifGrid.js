import { Fragment } from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";
import GifGridItem from './GifGridItem';
import PropTypes from 'prop-types';

export const GifGrid = ({category}) => {

  const {data: images, loading}= useFetchGifs(category)

  return ( 
    <Fragment>
      <h3>{category}</h3>
      {loading && 'Cargando...'}
      <div className='card-grid'>
        {images.map(img => (
          <GifGridItem
            key={img.id}
            {...img}
          />
        ))}
      </div>

    </Fragment>
   );
}
GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}

export default GifGrid;