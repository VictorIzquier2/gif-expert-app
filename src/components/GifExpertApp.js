import React, { Fragment, useState } from 'react';
import AddCategory from './AddCategory';
import GifGrid from './GifGrid';

const GifExpertApp = () => {

  //const categories = ['One Punch', 'Samurai X', 'Dragon Ball']
  const [categories, setCategories] = useState(
    ['']
  )

  return (
    <Fragment>
      <h2>GifExpertApp</h2>
      <AddCategory
        setCategories={setCategories}
      />
      <hr/>

      <ol>
        {categories.map(category => (
          <GifGrid 
            category={category}
            key={category}
          />
        ))}
      </ol>
    </Fragment>
   );
}
 
export default GifExpertApp;