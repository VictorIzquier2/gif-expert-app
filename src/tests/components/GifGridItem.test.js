import React from 'react';
import {shallow} from 'enzyme';
import {GifGridItem} from '../../components/GifGridItem';

describe('Pruebas en <GifGridItem />', () => {
  
  const title = 'Goku comiendo';
  const url = 'https://media3.giphy.com/media/DzeKll1HYxiYo/giphy.gif?cid=f18a8f9bn8frrg5r778mjk79cowvlr8ixxmqp121maxk18bz&rid=giphy.gif'
  const wrapper = shallow(<GifGridItem title={title} url={url} />)

  test('Debe de mostrar el componente correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  })

  test('Debe de tener un párrafo con el título', () => {
    
    const p = wrapper.find('p');
    expect(p.text().trim()).toBe(title);
  });

  test('Debe de tener la imagen igual al url y alt de los props', () => {
    const img = wrapper.find('img');
    expect(img.prop('src')).toBe(url);
    expect(img.prop('alt')).toBe(title);
  })

});