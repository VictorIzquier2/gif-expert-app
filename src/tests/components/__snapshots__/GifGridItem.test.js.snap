// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> Debe de mostrar el componente correctamente 1`] = `
<Fragment>
  <div
    className="card"
  >
    <img
      alt="Goku comiendo"
      height={240}
      src="https://media3.giphy.com/media/DzeKll1HYxiYo/giphy.gif?cid=f18a8f9bn8frrg5r778mjk79cowvlr8ixxmqp121maxk18bz&rid=giphy.gif"
    />
    <p>
      Goku comiendo
    </p>
  </div>
</Fragment>
`;
